solutions/silver/cf-1879D.mdx

```diff
@@ -0,0 +1,91 @@
+---
+id: cf-1879D
+source: CF 1879D
+title: Sum of XOR Functions
+author: Justin Ji
+---
+
+[Official Analysis (C++)](https://codeforces.com/blog/entry/120773)
+
+<Spoiler title = "Hint #1">
+
+Consider processing the expression bit by bit, and evaluating the contribution
+of each bit in the answer.
+
+</Spoiler>
+
+<Spoiler title = "Hint #2">
+
+The contribution of each bit is tied to the sum of lengths of all subarrays that
+have $f(l, r)$ include this bit. When would a subarray have its value of
+$f(l, r)$ include a given bit?
+
+</Spoiler>
+
+<Spoiler title = "Explanation">
+
+## Explanation
+
+Let's try to simplify the problem. Consider processing the array bit by bit.
+The $x$'th least significant bit contributes $2^x \cdot \sum_{l=1}^{n} \sum_{r=l}^{n} p \cdot
+(r - l + 1)$ to the answer, where $p$ is the parity of this bit's occurrences.
```

Comment:
i mean this is redundant now given this hints

Reply:
removed that first part